flutter/lib/src/default_integrations.dart

```diff
@@ -140,3 +142,13 @@ Integration nativeSdkIntegration(SentryOptions options, MethodChannel channel) {
 
   return integration;
 }
+
+/// Integration that captures certain window and device events.
+/// See also:
+///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+void widgetsBindingIntegration(Hub hub, SentryOptions options) {
+  // We don't need to call `WidgetsFlutterBinding.ensureInitialized()`
+  // because `FlutterSentry.init` already calls it.
+  WidgetsBinding.instance.addObserver(SentryWidgetsBindingObserver(hub: hub));
+  options.sdk.addIntegration('widgetsBindingIntegration');
+}
```

Comment:
as this is an automatic integration and people may want to turn it off, it'd be nice to expose an option to do so, see: 

https://github.com/getsentry/sentry-dart/blob/main/dart/lib/src/sentry_options.dart#L243-L252

but creating a new bool as this is Flutter's specific, and then before adding the observer to the WidgetsBinding.instance, we check this bool which is enabled by default.

see an example:

https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/main/java/io/sentry/android/core/ActivityBreadcrumbsIntegration.java#L45-L48